src/commands/install.rs

```diff
@@ -43,29 +51,64 @@ impl InstallSubcommand {
         let mut package_sources = PackageSourceMap::new(default_registry);
         package_sources.add_fallbacks()?;
 
-        let mut try_to_use = BTreeSet::new();
-        for package in lockfile.packages {
-            match package {
-                LockPackage::Registry(registry_package) => {
-                    try_to_use.insert(PackageId::new(
-                        registry_package.name,
-                        registry_package.version,
-                    ));
-                }
-                LockPackage::Git(_) => {}
+        let try_to_use = lockfile.as_ids().collect();
+
+        let progress = ProgressBar::new(0).with_style(
+            ProgressStyle::with_template("{spinner:.cyan}{wide_msg}")?.tick_chars("⠁⠈⠐⠠⠄⠂ "),
+        );
+
+        progress.enable_steady_tick(Duration::from_millis(100));
+
+        if self.locked {
+            progress.println(format!(
+                "{} Verifying {}lockfile is up-to-date...",
+                SetForegroundColor(Color::DarkGreen),
+                SetForegroundColor(Color::Reset)
+            ));
+
+            let latest_graph = resolve(&manifest, &BTreeSet::new(), &package_sources)?;
+
+            if try_to_use != latest_graph.activated {
+                progress.finish_and_clear();
+
+                let old_dependencies = &try_to_use;
+
+                let changes = generate_depedency_changes(old_dependencies, &latest_graph.activated);
+                let mut error_output = Vec::new();
+
+                writeln!(
+                    error_output,
+                    "{} The Lockfile is out of date and wasn't changed due to --locked{}",
+                    SetForegroundColor(Color::Yellow),
+                    SetForegroundColor(Color::Reset)
+                )?;
+                render_update_difference(&changes, &mut error_output)?;
+                writeln!(
+                    error_output,
+                    "{}{} Suggestion{}{} try running wally update",
+                    SetAttribute(Attribute::Bold),
+                    SetForegroundColor(Color::DarkGreen),
+                    SetForegroundColor(Color::Reset),
+                    SetAttribute(Attribute::Reset)
+                )?;
+
+                // Should be safe since we're only getting valid utf-8.
+                anyhow::bail!(String::from_utf8(error_output).unwrap());
             }
```

Comment:
```suggestion
                anyhow::bail!(String::from_utf8(error_output)
                    .expect("output from render_update_difference should always be utf-8"));
```
`expect` is better if you expect it to never fail. `unwrap` is more of a "I don't care if it fails".

Reply:
Oh good, VSCode messed up the alignment on this one.

Hopefully you know what I meant.

Reply:
I did lol